primefaces/src/main/java/org/primefaces/component/selectoneradio/SelectOneRadio.java

```diff
@@ -47,4 +47,24 @@ public class SelectOneRadio extends SelectOneRadioBase {
     public String getRadioButtonId(FacesContext context, int index) {
         return this.getClientId(context) + UINamingContainer.getSeparatorChar(context) + index;
     }
+
+    @Override
+    public String getInputClientId() {
+        return getRadioButtonId(FacesContext.getCurrentInstance(), 0);
```

Comment:
getFacesContext() from UIComponentBase
in MF its cached one vs ThreadLocal lookup

Reply:
Good catch!  Fixed